cmd/push_test.go

```diff
@@ -86,114 +46,46 @@ func Test_SyncOK(t *testing.T) {
 }
 
 func Test_SyncNoURL(t *testing.T) {
-	ts := NewTestStore(t, "O")
-	defer ts.Done(t)
-
+	_, _, okp := CreateOperatorKey(t)
+	as, m := RunTestAccountServerWithOperatorKP(t, okp)
+	ts := NewTestStoreWithOperatorJWT(t, string(m["operator"]))
 	ts.AddAccount(t, "A")
+	as.Close()
 
-	// create an http server to accept the request
-	hts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		defer r.Body.Close()
-
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			require.NoError(t, err)
-		}
-		_, err = jwt.DecodeAccountClaims(string(body))
-		require.NoError(t, err)
-
-		err = ts.Store.StoreClaim(body)
-		require.NoError(t, err)
-
-		w.WriteHeader(200)
-	}))
-	defer hts.Close()
+	// remove the account server so we cannot push
+	oc, err := ts.Store.ReadOperatorClaim()
+	require.NoError(t, err)
+	oc.AccountServerURL = ""
+	token, err := oc.Encode(okp)
+	require.NoError(t, err)
+	ts.Store.StoreClaim([]byte(token))
 
-	// sync the store
-	_, _, err := ExecuteCmd(createPushCmd(), "--account", "A")
+	_, _, err = ExecuteCmd(createPushCmd(), "--account", "A")
 	require.Error(t, err)
-	require.Contains(t, err.Error(), "no account server url was provided")
+	t.Log(err.Error())
+	require.Contains(t, err.Error(), "no account server url was provided by the operator jwt")
 }
 
 func Test_SyncNoServer(t *testing.T) {
-	ts := NewTestStore(t, "O")
-	defer ts.Done(t)
-
+	as, m := RunTestAccountServer(t)
+	ts := NewTestStoreWithOperatorJWT(t, string(m["operator"]))
 	ts.AddAccount(t, "A")
+	as.Close()
 
-	// create an http server to accept the request
-	hts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		defer r.Body.Close()
-
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			require.NoError(t, err)
-		}
-		_, err = jwt.DecodeAccountClaims(string(body))
-		require.NoError(t, err)
-
-		err = ts.Store.StoreClaim(body)
-		require.NoError(t, err)
-
-		w.WriteHeader(200)
-	}))
-	setOperatorAccountServer(t, ts, hts, nil)
-	hts.Close()
-
-	// sync the store
 	_, stderr, err := ExecuteCmd(createPushCmd(), "--account", "A")
 	require.Error(t, err)
 	require.Contains(t, stderr, "connect: connection refused")
 }
 
 func Test_SyncManaged(t *testing.T) {
-	_, pk, kp := CreateOperatorKey(t)
-	oc := jwt.NewOperatorClaims(pk)
-	oc.Name = "O"
-	d, err := oc.Encode(kp)
-	require.NoError(t, err)
+	as, m := RunTestAccountServer(t)
+	defer as.Close()
 
-	ts := NewTestStoreWithOperatorJWT(t, d)
+	ts := NewTestStoreWithOperatorJWT(t, string(m["operator"]))
```

Comment:
i wonder if it will be more maintainable if this function just takes the map so you don't need the magic string everywhere

Reply:
it is a test, and the "operator" corresponds to id of the operator.

Reply:
i know, just saying it is a "test account server" -> "test operator store" which i saw a bunch of places and has a magic string